### ISSUE TITLE
Radex RD1212v1 - not found

### ISSUE THREAD

**Issue by ZeroChaos-** [luigifab/python-radexreader#8]
```
zero@theprophet radexreader % sudo radexreader readall                                                                                                       (git)-[master] 
Information   python3-radexreader 1.2.1 with python 3.10.5 + pyusb 1.2.1 + pyserial 3.5
Traceback (most recent call last):
  File "/usr/bin/radexreader", line 123, in <module>
    reader = radexreader.RadexReader()
  File "/usr/lib/python3.10/site-packages/radexreader/__init__.py", line 71, in __init__
    raise ValueError('Error: RADEX RD1212 (serial) not found')
ValueError: Error: RADEX RD1212 (serial) not found
```
it shows in lsusb
```
Bus 001 Device 024: ID 10c4:ea60 Silicon Labs CP210x UART Bridge
```
it shows in dmesg as well
```
[282336.627034] usb 1-1.3: new full-speed USB device number 24 using xhci_hcd
[282336.718944] usb 1-1.3: New USB device found, idVendor=10c4, idProduct=ea60, bcdDevice= 1.00
[282336.718948] usb 1-1.3: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[282336.718950] usb 1-1.3: Product: CP2102 USB to UART Bridge Controller
[282336.718951] usb 1-1.3: Manufacturer: Silicon Labs
[282336.718952] usb 1-1.3: SerialNumber: 0001
[282337.901111] usbcore: registered new interface driver cp210x
[282337.901121] usbserial: USB Serial support registered for cp210x
[282337.901185] cp210x 1-1.3:1.0: cp210x converter detected
[282337.904024] usb 1-1.3: cp210x converter now attached to ttyUSB0
```
Not sure what's wrong, any ideas?

**Comment by ZeroChaos-** [comment]
Based on some manual invocation or serial.tools.list_ports I think https://github.com/luigifab/python-radexreader/blob/master/src/radexreader/__init__.py#L66 is not right.
```
zero@theprophet radexreader % python -m serial.tools.list_ports RADEX                                                                                        
Filtered list with regexp: 'RADEX'
no ports found
zero@theprophet radexreader % python -m serial.tools.list_ports -v                                                                                           
/dev/ttyACM0        
    desc: USB Single Serial
    hwid: USB VID:PID=1A86:55D4 SER=5426004892 LOCATION=1-1.1.3:1.0
/dev/ttyUSB0        
    desc: CP2102 USB to UART Bridge Controller - CP2102 USB to UART Bridge Controller
    hwid: USB VID:PID=10C4:EA60 SER=0001 LOCATION=1-1.3
2 ports found
``` 
If I change the 'RADEX' grep to 'CP2102' it finds the device and fails later
```
Traceback (most recent call last):
  File "/usr/bin/radexreader", line 125, in <module>
    measures = reader.read(False)
  File "/usr/lib/python3.10/site-packages/radexreader/__init__.py", line 187, in read
    hexa = self.hid_get_report()
  File "/usr/lib/python3.10/site-packages/radexreader/__init__.py", line 170, in hid_get_report
    return self.serial.read(14)
  File "/usr/lib/python3.10/site-packages/serial/serialposix.py", line 595, in read
    raise SerialException(
serial.serialutil.SerialException: device reports readiness to read but returned no data (device disconnected or multiple access on port?)
```

**Comment by ZeroChaos-** [comment]
Interestingly, if I run over and over again it works sometimes, and then fails other times
```
zero@theprophet radexreader % sudo radexreader tail                                                                                                    (git)-[master]  255 ⨯
Information   python3-radexreader 1.2.1 with python 3.10.5 + pyusb 1.2.1 + pyserial 3.5
Manufacturer  0x10c4  Silicon Labs
Product       0xea60  CP2102 USB to UART Bridge Controller
ComPort       /dev/ttyUSB0

[info] Sensor: Geiger-Müller tube SBM 20-1
[info] Measuring range: 0.05 - 999 µSv/h
[info] Uncertainty of the result: ±(15+6/D)%
[info]   D is the value after a complete cycle of 100 seconds

[warn] The values displayed are only an approximation.
[warn] The actual value can only be measured with suitable professional device.
[warn] For now, not tested with measured values greater than 0.25 µSv/h.

2021-09-02 15:13:22    0.19 µSv/h  ±46% (0.10 ≤ 0.19 ≤ 0.28)
Traceback (most recent call last):
  File "/usr/bin/radexreader", line 58, in <module>
    measures = reader.read(True)
  File "/usr/lib/python3.10/site-packages/radexreader/__init__.py", line 186, in read
    if hexa[0] != 0:
IndexError: index out of range
```
With a variety of errors
```
zero@theprophet radexreader % sudo radexreader readlast                                                                                                      (git)-[master] 
Information   python3-radexreader 1.2.1 with python 3.10.5 + pyusb 1.2.1 + pyserial 3.5
Manufacturer  0x10c4  Silicon Labs
Product       0xea60  CP2102 USB to UART Bridge Controller
ComPort       /dev/ttyUSB0

[info] Sensor: Geiger-Müller tube SBM 20-1
[info] Measuring range: 0.05 - 999 µSv/h
[info] Uncertainty of the result: ±(15+6/D)%
[info]   D is the value after a complete cycle of 100 seconds

[warn] The values displayed are only an approximation.
[warn] The actual value can only be measured with suitable professional device.
[warn] For now, not tested with measured values greater than 0.25 µSv/h.

Traceback (most recent call last):
  File "/usr/bin/radexreader", line 96, in <module>
    measures = reader.read(True)
  File "/usr/lib/python3.10/site-packages/radexreader/__init__.py", line 185, in read
    hexa = self.hid_get_report()
  File "/usr/lib/python3.10/site-packages/radexreader/__init__.py", line 168, in hid_get_report
    return self.serial.read(14)
  File "/usr/lib/python3.10/site-packages/serial/serialposix.py", line 595, in read
    raise SerialException(
serial.serialutil.SerialException: device reports readiness to read but returned no data (device disconnected or multiple access on port?)
```

**Comment by luigifab** [comment]
For serial not found, you are right, for your device we need to grep on `CP2102`.
For `device reports readiness to read but returned no data`, I'm not sure, have you another program that try to access the device? or like RADEX ONE this is not the same baudrate, can you try `baudrate=9600`?

For `IndexError: index out of range`, we need to debug the `hexa` variable:

https://github.com/luigifab/python-radexreader/blob/fd77d6900155618d6ed189b0c7ddd0d48ff309a2/src/radexreader/__init__.py#L184-L186

I suppose that your device doesn't use the same addresses.

**Comment by ZeroChaos-** [comment]
Thanks for getting back to me.  I got two outputs back that are "hexa[0] != 0", I simply added "print(hexa) on the line above the if

b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b''

You were 100% right about me having another process running on the port, it was gpsd.  I stopped gpsd, restarted the radex, and reran several times *with no change to any of the reported issues, hitting them all again*.  My personal favorite, is that after I ran `radexreader erase` and then waited a day, this is the output of `radex readall`:
```
zero@theprophet ~ % sudo radexreader readall                                                                                                                             1 ⨯
Information   python3-radexreader 1.2.1 with python 3.10.5 + pyusb 1.2.1 + pyserial 3.5
Manufacturer  0x10c4  Silicon Labs
Product       0xea60  CP2102 USB to UART Bridge Controller
ComPort       /dev/ttyUSB0

[info] Sensor: Geiger-Müller tube SBM 20-1
[info] Measuring range: 0.05 - 999 µSv/h
[info] Uncertainty of the result: ±(15+6/D)%
[info]   D is the value after a complete cycle of 100 seconds

[warn] The values displayed are only an approximation.
[warn] The actual value can only be measured with suitable professional device.
[warn] For now, not tested with measured values greater than 0.25 µSv/h.

b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
b'\x00\x12\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
no data stored
```
I don't know if I messed up some config, or if erase was not a good idea, or what.

**Comment by ZeroChaos-** [comment]
Okay, I figured out why I wasn't getting any data at all since the erase.  I'm still getting the two weird strings, but now I'm also getting data.
I had the data collection set to "24 hours", which I now suspect means it logs once an hour, meaning there is nothing for you to poll for until it's run a full 24 hours.  When I set it to one minute, I get one successful read per minute (using `radexreader tail`) and all the reads in between are b'' (empty).  I can send a small pr with the grep and ignore empty changes after I test a little longer.  Or you can make the changes yourself, I'm okay either way.  I'm not a python coder, I just added a line to ignore empty reports
```
if hexa == b'':
  continue
```

**Comment by luigifab** [comment]
Yes, you are right, you need to configure the device to 1 min to get one read per minute, or 10 min to get one read every 10 minutes. I forget to add that in readme.

I will also try the `continue`.
You can make a PR, as you want, in both cases I will add the changes for next release.

**Comment by luigifab** [comment]
And with 24 hours, I think you will have one log for 24 hours. So you need to wait 24 hours to avoid `no data stored`.

**Comment by ZeroChaos-** [comment]
Do you have one of this model for testing, or do you need me to do it?  I can try to send a PR later today, especially if you don't have one for testing.

**Comment by luigifab** [comment]
I have only my RD1212 v2.48.
I testing with:
```python
[...]
			devices = serial.tools.list_ports.grep('RADEX')
			for device in devices:
				self.serial = serial.Serial(port=device.device, baudrate=115200, timeout=0.5)
				return # device found
			devices = serial.tools.list_ports.grep('CP2102')
			for device in devices:
				self.serial = serial.Serial(port=device.device, baudrate=115200, timeout=0.5)
				return # device found
[...]
			if hexa != b'' and hexa[0] != 0:
[...]
```

**Comment by luigifab** [comment]
About 1 min for data collection, I tried 5 min, and nothing works!
```
radexreader.py tail
[5 min]
2022-07-20 18:43:14  655.48 µSv/h  ±15% (557.10 ≤ 655.48 ≤ 753.86)
2022-07-20 18:48:14  655.48 µSv/h  ±15% (557.10 ≤ 655.48 ≤ 753.86)
[1 min]
2022-07-20 18:49:41    0.11 µSv/h  ±69% (0.03 ≤ 0.11 ≤ 0.19)
2022-07-20 18:50:41    0.11 µSv/h  ±69% (0.03 ≤ 0.11 ≤ 0.19)
```

**Comment by luigifab** [comment]
What is your exact device version? (if you have the information for example in menu/services).

**Comment by ZeroChaos-** [comment]
Version 2.03

**Comment by luigifab** [comment]
I don't remember, but perhaps fixed in v1.2.2.

**Comment by ZeroChaos-** [comment]
confirmed fixed in 1.2.2, thanks!
